scidbpy/interface.py

```
@@ -268,8 +268,10 @@ def _show_array(self, name, **kwargs):
             query = "show('store({0}, {1})', 'afl')".format(name, tmp)
         else:
             query = "show({0})".format(name)
+            #The replace is added to the end in order to get rid of extra backslash when compression is added. 
+            result =  self._execute_query(query, **kwargs).replace('\\','')
 
-        return self._execute_query(query, **kwargs)
+        return result
```

Nice, we may be able to just remove the manual unzipping call in SciDB-Py in that case (it's always been a workaround). 

Before we do that, though, we should make sure that shim is actually sending a compressed payload (since another subtle bug is to request compression from an unsupported version of shim, which ignores the request and returns uncompressed data). 

There's a test script bundled with shim that uses curl + compression + gunzip, and that should fail loudly if compression isn't working (https://github.com/Paradigm4/shim/blob/master/tests/compressed_stream.sh). Can you confirm that that test passes on the environment you are using?